core/renderers/common/info.js

```diff
@@ -612,16 +628,30 @@ Blockly.blockRendering.RenderInfo.prototype.getSpacerRowHeight_ = function(
  * vertically, with no special cases.  You will likely need extra logic to
  * handle (at minimum) top and bottom rows.
  * @param {!Blockly.blockRendering.Row} row The row containing the element.
- * @param {!Blockly.blockRendering.Measurable} _elem The element to place.
+ * @param {!Blockly.blockRendering.Measurable} elem The element to place.
  * @return {number} The desired centerline of the given element, as an offset
  *     from the top left of the block.
  * @protected
  */
 Blockly.blockRendering.RenderInfo.prototype.getElemCenterline_ = function(row,
-    _elem) {
-  var result = row.yPos;
-  result += (row.height / 2);
-  return result;
+    elem) {
```

Comment:
Next step: check if the other renderers can have their own logic and then just call the superclass function in the default case (at least for thrasos, maybe also for geras)